pepper-apis/docs/specification/src/endpoints/user.studies.activity.uploads.yml

```
@@ -0,0 +1,106 @@
+parameters:
+  - $ref: '../pepper.yml#/components/parameters/studyGuid'
+  - $ref: '../pepper.yml#/components/parameters/userGuid'
+  - $ref: '../pepper.yml#/components/parameters/instanceGuid'
+
+post:
+  operationId: postActivityUploads
+  tags:
+    - Activities
+  summary: create an authorized file upload
+  description: |
+    Request for a new URL for file upload. Link is accessible for limited
+    period of time (e.g. 5 mins). Resumable uploads are supported. With
+    information in response body, client will be able to upload the file
+    directly to a Google Bucket. If resumable=true is specified in request,
+    client will assume responsibility for creating the session URI with
+    provided URL in addition to follow up requests that contain the file
+    contents (see [documentation][doc]).
+
+    **Study Admins**
+
+    Activity file upload is not supported if activity instance/question is
+    read-only or otherwise inaccessible, unless operator is a study admin.
+
+    **Temporary Users**
+
+    This API endpoint does not allow access by temporary users.
+
+    [doc]: https://cloud.google.com/storage/docs/performing-resumable-uploads
+  requestBody:
+    required: true
+    content:
+      application/json:
+        schema:
+          type: object
+          required:
+            - questionStableId
+            - fileName
+            - fileSize
+            - resumable
+          properties:
+            questionStableId:
+              type: string
+              description: The question identifier
+            fileName:
+              type: string
+              maxLength: 255
+              description: Name of file to be uploaded
+            fileSize:
+              type: integer
+              minimum: 1
+              description: Size of the file, in bytes (subject to a limit such as 2MB)
```

Let's avoid mentioning a specific size here.

```suggestion
              description: Size of the file, in bytes. The server may limit the maximum size of uploaded files.
```